demos/ssh-demo/README.md

```diff
@@ -0,0 +1,41 @@
+# SSH demo
+
+To demonstrate confidential containers capabilities, we run a pod with SSH public key authentication.
```

Comment:
Can we elaborate a little more here on the merits of such demo?

For example by explaining that the container image will be encrypted and that neither the guest nor the host will hold the encryption keys. That the only way for this container image to be run and for anyone to be able to ssh into it will be by having the guest getting the encryption keys from a remote attestation server.

Although It *may* be obvious to us, it may not be so for anyone going through that README?

Reply:
Yes, this deserves some more background info. I just didn't want to write a ton before getting some degree of approval about doing this in the first place, but I guess we have that now :)

Reply:
I guess it is semantics if it is truely the attestation server that is used to get the keys v a further key broker service?

I would hesitate to prescribe the key must come from a remote attestation server since with at least one TEE technology (Secure Execution) it is possible to secure the key inside a guest image without needing to retrieve it from the attestation flow.
Indicating that the encryption key needs to be protected from the host I completely agree with but I feel protected from the guest needs to be discussed more. 
Fundamentally containers will run inside the guest and we don't claim to protect what is in the guest from the containers (compared to guest from host) , so if my guest image can hold the key in uncompromised way then I must have built the image for it to have the appropriate key hence attestation flow is not as critical. A guest image without the key won't be able to run the container image.
So:
If I build a dedicated guest image which is itself encrypted and provides protection then putting keys in there is reasonable.
If I build a generic guest image then this would raise concerns on how generic and start to strongly suggest attestation.
And of course if I cannot use encrypted guest images then attestation flow becomes our delivery mechanism.

Reply:
Well at the very least we should probably try to explain the difference between using exec to access the container and using SSH with this setup. We are creating a secure channel between the guest owner and the workload. This is mediated by a trusted entity. At this level there is really no difference between using a KBC that gets the secret via SEV secret injection and a KBC that gets the secret via an encrypted initrd. Those are two different secure mechanisms for provisioning the secrets that trust a non-csp entity. In general it might be good to focus on how these platforms are fundamentally aligned.